### PR TITLE
New package: Oven-sh.Bun version 1.1.0

### PR DESCRIPTION
**Pull request by mdanish-kh**
Checklist for Pull Requests
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/147222)

### CODE REVIEW

**Comment by sitiom** [comment]
- Related to https://github.com/oven-sh/bun/pull/9770

**Review by sitiom — COMMENTED** [review]
As suggested by @paperdave, a CPU check like this would be nice, but I guess it's not possible with Winget yet:
```powershell
if (-not (Add-Type -MemberDefinition '[DllImport("kernel32.dll")] public static extern bool IsProcessorFeaturePresent(int ProcessorFeature);'-Name 'Kernel32'-Namespace 'Win32'-PassThru)::IsProcessorFeaturePresent(40)) {
    Write-Host -Foreground Red "This CPU does not support the AVX2 instruction set. Please install 'Oven-sh.Bun.Baseline' instead."
}
```

**Comment by mdanish-kh** [comment]
> As suggested by @paperdave, a CPU check like this would be nice, but I guess it's not possible with Winget yet:
> 
> ```powershell
> if (-not (Add-Type -MemberDefinition '[DllImport("kernel32.dll")] public static extern bool IsProcessorFeaturePresent(int ProcessorFeature);'-Name 'Kernel32'-Namespace 'Win32'-PassThru)::IsProcessorFeaturePresent(40)) {
>     Write-Host -Foreground Red "This CPU does not support the AVX2 instruction set. Please install 'Oven-sh.Bun.Baseline' instead."
> }
> ```

Yes, I don't think that's possible with WinGet. We can have a note printed out after installation though that says something like:

`If you face issues running the package, it might be because your CPU does not support AVX2 instruction set. In that case, try installing Over-sh.Bun.Baseline instead.`

File: manifests/o/Oven-sh/Bun/1.1.0/Oven-sh.Bun.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+PackageVersion: 1.1.0
+InstallerType: zip
+NestedInstallerType: portable
+NestedInstallerFiles:
+- RelativeFilePath: bun-windows-x64\bun.exe
+  PortableCommandAlias: bun
+Installers:
+- Architecture: x64
+  InstallerUrl: https://github.com/oven-sh/bun/releases/download/bun-v1.1.0/bun-windows-x64.zip
+  InstallerSha256: 60E9291D09657AB315BCC6BC07CBF67D2E9B9B18C21BB7CF9CE2D9FA983C9719
```
Comment (sitiom) [inline manifests/o/Oven-sh/Bun/1.1.0/Oven-sh.Bun.installer.yaml:14]:
```suggestion
  InstallerSha256: 60E9291D09657AB315BCC6BC07CBF67D2E9B9B18C21BB7CF9CE2D9FA983C9719
  Dependencies:
    PackageDependencies:
    - PackageIdentifier: Microsoft.VCRedist.2015+.x64
```

**Review by sitiom — CHANGES_REQUESTED** [review]

**Review by russellbanks — APPROVED** [review]

File: manifests/o/Oven-sh/Bun/1.1.0/Oven-sh.Bun.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Created using wingetcreate 1.6.1.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.6.0.schema.json
+
+PackageIdentifier: Oven-sh.Bun
+PackageVersion: 1.1.0
+InstallerType: zip
+NestedInstallerType: portable
+NestedInstallerFiles:
+- RelativeFilePath: bun-windows-x64\bun.exe
+  PortableCommandAlias: bun
```
Comment (sitiom) [inline manifests/o/Oven-sh/Bun/1.1.0/Oven-sh.Bun.installer.yaml:10]: I forgot about this too, a `bunx` alias for `bun x` (Bun automatically detects the naming):
```suggestion
  PortableCommandAlias: bun
- RelativeFilePath: bun-windows-x64\bun.exe
  PortableCommandAlias: bunx
```

Comment (sitiom) [reply manifests/o/Oven-sh/Bun/1.1.0/Oven-sh.Bun.installer.yaml:10]: - https://github.com/microsoft/winget-pkgs/pull/147295

**Pull request merged**